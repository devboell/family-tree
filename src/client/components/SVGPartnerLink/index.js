import React from 'react'
import PropTypes from 'prop-types'

const SVGPartnerLink = ({
  originX,
  originY,
  offsetX,
  previousSubTreeWidth,
  dimensions,
}) => {
  const { boxWidth, marginX } = dimensions
  const targetX = (
    (previousSubTreeWidth &&
      (offsetX - (previousSubTreeWidth / 2)) + boxWidth + marginX
    )
    || originX - marginX
  ) + 2

  return (
    <path stroke="red" d={`M ${originX} ${originY} H ${targetX}`} />
  )
}


SVGPartnerLink.propTypes = {
  originX: PropTypes.number.isRequired,
  originY: PropTypes.number.isRequired,
  offsetX: PropTypes.number.isRequired,
  previousSubTreeWidth: PropTypes.number.isRequired,
  dimensions: PropTypes.shape({}).isRequired,
}


export default SVGPartnerLink
