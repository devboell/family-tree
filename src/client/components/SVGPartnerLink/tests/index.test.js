import React from 'react'
import SVGPartnerLink from '../index'

let props = {
  originX: 300,
  originY: 100,
  offsetX: 50,
  previousSubTreeWidth: 200,
  dimensions: { boxWidth: 80, marginX: 50 },
}


it('SVGPartnerLink with previousSubTreeWidth, renders component', () => {
  const wrapper = shallow(<SVGPartnerLink {...props} />)
  expect(wrapper).toMatchSnapshot()
})


it('SVGPartnerLink without previousSubTreeWidth, renders component', () => {
  props = {
    ...props,
    previousSubTreeWidth: 0,
  }
  const wrapper = shallow(<SVGPartnerLink {...props} />)
  expect(wrapper).toMatchSnapshot()
})
