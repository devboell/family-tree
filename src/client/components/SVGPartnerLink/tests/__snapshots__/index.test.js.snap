// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGPartnerLink with previousSubTreeWidth, renders component 1`] = `
<path
  d="M 300 100 H 82"
  stroke="red"
/>
`;

exports[`SVGPartnerLink without previousSubTreeWidth, renders component 1`] = `
<path
  d="M 300 100 H 252"
  stroke="red"
/>
`;
